test/lit.cfg.py

```diff
@@ -56,6 +56,8 @@
 
 # Tweak the PATH to include the tools dir.
 llvm_config.with_environment('PATH', config.llvm_tools_dir, append_path=True)
+# Tweak the PATH to include the binary build dir, in order to pick up CAPI tests during out-of-tree.
+llvm_config.with_environment('PATH', os.path.join(config.llvm_build_dir, 'bin'), append_path=True)
```

Comment:
I'm not sure if this is the right thing to do but this (and `config.llvm_build_dir = "@CMAKE_BINARY_DIR@"` below) is the only way I could get lit to find the `torch-mlir-capi-torch-test` executable (which gets deposited there). 

Note that

```
config.torch_mlir_obj_root = "@TORCH_MLIR_BINARY_DIR@"
...
llvm_config.with_environment('PATH', config.torch_mlir_obj_root, append_path=True)
```

doesn't work because `TORCH_MLIR_BINARY_DIR==CMAKE_CURRENT_BINARY_DIR==torch-mlir/build/tools/torch-mlir` for out-of-tree.

Finally 

```
add_lit_testsuite(check-torch-mlir-capi "Running the torch-mlir CAPI tests"
        ${CMAKE_CURRENT_BINARY_DIR}
        --->EXCLUDE_FROM_CHECK_ALL<---
        DEPENDS torch-mlir-capi-torch-test
        )
```

doesn't work either (with or without 

```
add_dependencies(check-torch-mlir-all
  check-torch-mlir
  check-torch-mlir-dialects
  check-torch-mlir-capi
  )
```

Reply:
Hmmm... I'm not an expert but this seems reasonable.